frontend/src/lib/k8s/apiProxy.ts

```diff
@@ -98,7 +99,7 @@ export async function request(
 
     const error: Error & { status?: number } = new Error(message);
     error.status = status;
-    throw error;
+    return Promise.reject(error);
```

Comment:
Why is this need? And doesn't this require changing the code calling/catching this function?

Reply:
With throw we have problems of not being able to handle the errors 
There are two solutions to it either wrap the entire function in a promise and then throw from it or return an error wrapped in Promise.reject so that our catch can handle it
See https://stackoverflow.com/a/33446005/8527880

Reply:
What I mean is that we've been using it this way. So I imagine we'd have to change some of the calling code that may be catching the exception already.

Reply:
I checked i think we don't need to change other parts of code for this as at all places we use the .catch method on promise